beacon_node/lighthouse_network/src/service/mod.rs

```diff
@@ -1858,9 +1858,11 @@ impl<E: EthSpec> Network<E> {
                         None
                     }
                     RpcSuccessResponse::MetaData(meta_data) => {
-                        self.peer_manager_mut()
+                        let updated_cgc = self
+                            .peer_manager_mut()
                             .meta_data_response(&peer_id, meta_data.as_ref().clone());
-                        None
+                        // Send event after calling into peer_manager so the PeerDB is updated
+                        self.build_response(id, peer_id, Response::MetaData(meta_data, updated_cgc))
```

Comment:
Found out why this doesn't work - `self.build_response` here always return `None` because the `AppRequestId` for `MetaData` is always `AppRequestId::Internal`:

https://github.com/sigp/lighthouse/blob/ba0b62ba175b45e6f050d0ce88c213262de87c50/beacon_node/lighthouse_network/src/service/mod.rs#L1387-L1398

Reply:
Addressed here:
https://github.com/sigp/lighthouse/pull/6975/commits/aab502f8a3d6eb17ea88b4ec09cd22472b916fdf